diff --git a/src/game/Chat.cpp b/src/game/Chat.cpp
index 58ed497..1ff348d 100644
--- a/src/game/Chat.cpp
+++ b/src/game/Chat.cpp
@@ -467,6 +467,7 @@ ChatCommand * ChatHandler::getCommandTable()
 
         { "aura",           SEC_ADMINISTRATOR,  false, &ChatHandler::HandleAuraCommand,                "", NULL },
         { "unaura",         SEC_ADMINISTRATOR,  false, &ChatHandler::HandleUnAuraCommand,              "", NULL },
-        { "announce",       SEC_MODERATOR,      true,  &ChatHandler::HandleAnnounceCommand,            "", NULL },
+        { "sysannounce",    SEC_MODERATOR,      true,  &ChatHandler::HandleSysAnnounceCommand,         "", NULL },
+        { "announce",       SEC_MODERATOR,      false,  &ChatHandler::HandleAnnounceCommand,            "", NULL },
         { "notify",         SEC_MODERATOR,      true,  &ChatHandler::HandleNotifyCommand,              "", NULL },
         { "goname",         SEC_MODERATOR,      false, &ChatHandler::HandleGonameCommand,              "", NULL },
diff --git a/src/game/Chat.h b/src/game/Chat.h
index fdbe565..0d3a7d1 100644
--- a/src/game/Chat.h
+++ b/src/game/Chat.h
@@ -106,6 +106,7 @@ class ChatHandler
         bool HandleGonameCommand(const char* args);
         bool HandleGroupgoCommand(const char* args);
         bool HandleRecallCommand(const char* args);
+        bool HandleSysAnnounceCommand(const char* args);
         bool HandleAnnounceCommand(const char* args);
         bool HandleNotifyCommand(const char* args);
         bool HandleGMmodeCommand(const char* args);
diff --git a/src/game/Language.h b/src/game/Language.h
index aee0f9b..63f6aa6 100644
--- a/src/game/Language.h
+++ b/src/game/Language.h
@@ -27,6 +27,7 @@ enum MangosStrings
 
     // level 0 chat
     LANG_SYSTEMMESSAGE                  = 3,
+    LANG_ANNOUNCE_COLOR                 = 2000,
     LANG_EVENTMESSAGE                   = 4,
     LANG_NO_HELP_CMD                    = 5,
     LANG_NO_CMD                         = 6,
diff --git a/src/game/Level1.cpp b/src/game/Level1.cpp
index 56e805f..98359dc 100644
--- a/src/game/Level1.cpp
+++ b/src/game/Level1.cpp
@@ -117,7 +117,7 @@ bool ChatHandler::HandleNpcWhisperCommand(const char* args)
 }
 
 // global announce
-bool ChatHandler::HandleAnnounceCommand(const char* args)
+bool ChatHandler::HandleSysAnnounceCommand(const char* args)
 {
     if(!*args)
         return false;
@@ -126,6 +126,15 @@ bool ChatHandler::HandleAnnounceCommand(const char* args)
     return true;
 }
 
+bool ChatHandler::HandleAnnounceCommand(const char* args)
+{
+    if(!*args)
+        return false;
+
+    sWorld.SendWorldText(LANG_ANNOUNCE_COLOR, m_session->GetPlayer()->GetName(), args);
+    return true;
+}
+
 //notification player at the screen
 bool ChatHandler::HandleNotifyCommand(const char* args)
 {
